test/FastTests/TestsInheritanceFastTests.cs

```diff
@@ -0,0 +1,154 @@
+            if (array.Length == numberToTolerate)
+                return;
+
+            var userMessage = $"We have detected '{array.Length}' test(s) that do not have {nameof(RavenFactAttribute)} or {nameof(RavenTheoryAttribute)} attribute. Please check if tests that you have added have those attributes. List of test files:{Environment.NewLine}{string.Join(Environment.NewLine, array.Select(x => GetTestName(x)))}";
+            throw new Exception(userMessage);
+
+            static string GetTestName(MethodInfo method)
+            {
+                return $"{method.DeclaringType?.FullName}.{method.Name}";
+            }
+
+            static bool Filter(MethodInfo method)
+            {
+                var factAttribute = method.GetCustomAttribute(typeof(FactAttribute), false);
+                if (factAttribute != null)
+                {
+                    if (ValidNamespace(factAttribute.GetType().Namespace))
+                        return false;
+
+                    return true;
+                }
+
+                var theoryAttribute = method.GetCustomAttribute(typeof(TheoryAttribute), false);
+                if (theoryAttribute != null)
+                {
+                    if (ValidNamespace(theoryAttribute.GetType().Namespace))
+                        return false;
+
+                    return true;
+                }
+
+                return false;
+            }
+
+            static bool ValidNamespace(string @namespace)
+            {
+                return @namespace == null || @namespace.StartsWith("FastTests") || @namespace.StartsWith("SlowTests") || @namespace.StartsWith("Tests.Infrastructure");
+            }
+        }
+
+        internal static IEnumerable<Assembly> GetAssemblies(HashSet<Assembly> assemblies, Assembly assemblyToScan)
+        {
+            if (assemblies.Add(assemblyToScan) == false)
+                yield break;
+
+            yield return assemblyToScan;
+
+            foreach (var asm in assemblyToScan.GetReferencedAssemblies())
+            {
+
+                Assembly load;
+                try
+                {
+                    load = Assembly.Load(asm);
+                }
+                catch
+                {
+                    continue;
+                }
+                foreach (var assembly in GetAssemblies(assemblies, load))
+                    yield return assembly;
+            }
+        }
+
+        internal static Type[] GetAssemblyTypes(Assembly assemblyToScan)
+        {
+            try
+            {
+                return assemblyToScan.GetTypes();
+            }
+            catch
+            {
+                return Array.Empty<Type>();
+            }
+        }
+
+        private static readonly string TargetDirectory = "test/SlowTests/";
+        private static string FindSlowTestsDirectory()
+        {
+            for (int i = 1; i < 10; i++)
+            {
+                var dir = string.Concat(Enumerable.Repeat("../", i)) + TargetDirectory;
```

Comment:
use `AppContext.BaseDirectory`, check what we do in EmbeddedTests

Reply:
done I also added `PathUtil.ToFullPath()`